python_modules/dagster/dagster/core/types/config.py

```diff
@@ -23,7 +23,7 @@ def __new__(cls, is_builtin=False, is_system_config=False, is_named=True):
 
 
 class ConfigType(object):
-    def __init__(self, name=None, type_attributes=DEFAULT_TYPE_ATTRIBUTES, description=None):
+    def __init__(self, key, name, type_attributes=DEFAULT_TYPE_ATTRIBUTES, description=None):
```

Comment:
I wonder if it'd be possible to autogenerate a key from the name within this function if one is not provided explicitly, rather than having all the callsites pass both the name and key (and usually as the same value)? Might give us a good place to implement a `name->key` function that isn't 1:1.